PathPlanning/DStarLite/d_star_lite.py

```diff
@@ -167,23 +187,25 @@ def compare_keys(self, key_pair1: tuple[float, float],
 
     def compute_shortest_path(self):
         self.U.sort(key=lambda x: x[1])
-        while (len(self.U) > 0 and
-               self.compare_keys(self.U[0][1],
-                                 self.calculate_key(self.start))) or \
-                self.rhs[self.start.x][self.start.y] != \
-                self.g[self.start.x][self.start.y]:
+        while (
+            len(self.U) > 0
+            and self.compare_keys(self.U[0][1], self.calculate_key(self.start))
+        ) or (
+            self.rhs[self.start.x, self.start.y]
+            != self.g[self.start.x, self.start.y]
+        ).any():
```

Comment:
Not this PR topic, but I think each while condition is challenging to read, so is it possible to introduce boolean variables or other kinds of intermediate values for readability?

Reply:
That is a great suggestion, thank you! I've updated this as you have suggested. Please take a look.